### ISSUE TITLE
Testing for default connection

### ISSUE THREAD

**Issue by ccoffrin** [dwavesystems/dwave-cloud-client#21]
Is there a way to test if a default connection is available in `.dwrc` before calling `dwave_micro_client.Connection()`?  If not, this is a feature request.

As of v0.2.2 calling `dwave_micro_client.Connection()` without a `.dwrc` for me results in,
```
...
  File "/usr/local/lib/python2.7/site-packages/dwave_micro_client.py", line 168, in __init__
    url, token, proxies, self.default_solver = load_configuration(url)
  File "/usr/local/lib/python2.7/site-packages/dwave_micro_client.py", line 1440, in load_configuration
    raise IOError("Could not load configuration from {}".format(file_path))
IOError: Could not load configuration from /Users/carleton/.dwrc
unable to parse stdin as a json document
```

**Comment by randomir** [comment]
Absolutely valid feature request. We're already doing something along those lines with the auth refactor that's in progress. :)

**Comment by randomir** [comment]
As of `v0.3.0` (Mar 8), configuration via `.dwrc` is deprecated in favor of `dwave.conf`.

[Introduction](http://dwave-cloud-client.readthedocs.io/en/latest/reference/intro.html) to the new config system is available in the docs. Detailed docs on how to use it under [`dwave.cloud.client.Client.from_config`](http://dwave-cloud-client.readthedocs.io/en/latest/reference/generated/dwave.cloud.client.Client.from_config.html#dwave.cloud.client.Client.from_config).

Now there are several ways/layers on which you can define the default solver: via config file - under solver section and under defaults section, via env var, and during call to `from_config` via kwargs.

Lastly, feature-based solver selection (i.e. not name/id-based) is also available in `v0.4.3+`. It's not yet visible on RTD, but it's [documented in docstring](https://github.com/dwavesystems/dwave-cloud-client/blob/e1af7628278858229012af59805ef9f52d21ba13/dwave/cloud/client.py#L608).

You can, for example, get the first QPU-based solver with at least 1000 qubits (but no upper bound) like this:
```
first_qpu_solver = client.solvers(qpu=True, num_qubits=[1000, None])[0]
```
